### PR TITLE
Improve error messages for why a solution failed to load

### LINKED ISSUE #5036: Improve error messages for why a solution file is invalid
We currently provide a generic message when a solution file fails to load. We should investigate various cases and error messages that we want to provide to the user.

### PR DESCRIPTION
**Pull request by jgoshi**
@piotrpMSFT @livarcocc @krwq @jonsequitur 

Fixes #5036

### CODE REVIEW

**Comment by jgoshi** [comment]
test RHEL7.2 x64 Release Build

File: src/Microsoft.DotNet.Cli.Sln.Internal/LocalizableStrings.cs [hunk 1]
```diff
@@ -2,6 +2,14 @@
 {
     internal class LocalizableStrings
     {
+        // {0} is the line number
+        // {1} is the error message details
+        public const string ErrorMessageFormatString = "Invalid format in line {0}: {1}";
```
Comment (livarcocc) [inline src/Microsoft.DotNet.Cli.Sln.Internal/LocalizableStrings.cs]: If you do this now, we will need to push this through Loc again.

@srivatsn Does this meet the bar for loc?

File: src/dotnet/CommonLocalizableStrings.cs [hunk 2]
```diff
@@ -105,7 +105,7 @@ internal class CommonLocalizableStrings
         public const string CouldNotFindSolutionIn = "Specified solution file {0} does not exist, or there is no solution file in the directory.";
         public const string CouldNotFindSolutionOrDirectory = "Could not find solution or directory `{0}`.";
         public const string MoreThanOneSolutionInDirectory = "Found more than one solution file in {0}. Please specify which one to use.";
-        public const string InvalidSolution = "Invalid solution `{0}`.";
+        public const string InvalidSolutionFormatString = "Invalid solution `{0}`. {1}"; // {0} is the solution path, {1} is already localized details on the failure
```
Comment (livarcocc) [inline src/dotnet/CommonLocalizableStrings.cs]: Same here.

File: src/Microsoft.DotNet.Cli.Sln.Internal/LocalizableStrings.cs (hunk 1 above)
Comment (srivatsn) [inline src/Microsoft.DotNet.Cli.Sln.Internal/LocalizableStrings.cs]: We can club this with the add package lot of strings

**Comment by livarcocc** [comment]
@srivatsn for approval.

**Pull request merged**